src/core/analyzer.ts

```diff
@@ -88,10 +88,12 @@ export class Analyzer {
       return await previousContext.analyze(this.brandUserInputUrls(urls));
     })();
     const context = await this._analysisComplete;
-    const results = new Map(this.brandUserInputUrls(urls).map(
-        (url) => [url, context.getDocument(context.resolveUrl(url))] as
-                     [string, Document | Warning]));
-    return new Analysis(results);
+    const resolvedUrls = context.resolveUrls(this.brandUserInputUrls(urls));
+    const results = new Map(resolvedUrls.map((url): [
```

Comment:
That type annotation is a little confusing to read to me in that position. How about casting the return value?

related: https://github.com/Microsoft/TypeScript/issues/10195

Reply:
I'd like to avoid the cast here, because a cast will turn a `string` into a `ResolvedUrl`, which we don't want here.